src/winstanley/WdlAnnotator.scala

```diff
@@ -36,6 +36,14 @@ class WdlAnnotator extends Annotator {
         }
       } else
         ()
+    case declaration: WdlDeclaration =>
+      if (declaration.getSetter == null)
+        psiElement.getRootElement match {
+          case _: WdlDraft3File =>
+            annotationHolder.createErrorAnnotation(psiElement, "Immediate assignment required for non-input declaration [draft-3]")
+          case _: WdlDraft2File =>
+            annotationHolder.createWeakWarningAnnotation(psiElement, "Non-input declarations will require immediate assignment in a future version of WDL")
```

Comment:
I can be talked out of this very easily, wanted to see what it would look like

Reply:
Yeah, I think I'd not include this... since the `input {}` section is not even allowed in draft-2, there's actually no way to make this go away